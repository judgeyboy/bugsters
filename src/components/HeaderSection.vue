<template>
  <div class="header-section">
    <img :src="require('@/assets/cute-bug.png')" width="150">
    <h1>{{ $t('message.welcome') }}</h1>
    <h2>Home of Professional Software Testers</h2>
    <a class="btn btn-primary" v-section-scroll location="about">Read More</a>
  </div>
</template>

<script>
export default {
  name: 'HeaderSection'
}
</script>

<style lang="sass" scoped>
.header-section
  margin-top: 4rem
  height: calc(100vh - 3.8rem)
  width: calc(100vw - 12rem)
  text-align: center

@media (max-width: 960px)
  .header-section
    width: 100vw
    margin-top: 0
</style>
