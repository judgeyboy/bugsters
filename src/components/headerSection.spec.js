import Vue from 'vue'
import { shallow } from 'vue-test-utils'
import SectionScroll from '@/plugins/section-scroll'

import HeaderSection from './HeaderSection'

Vue.use(SectionScroll)

jest.mock('@/assets/cute-bug.png', () => 'bugsters-logo.png')

describe('HeaderSection Component', () => {

  it('should be a Vue instance', () => {
    const wrapper = shallow(HeaderSection)
    expect(wrapper.isVueInstance).toBeTruthy()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(HeaderSection)
    const $html = wrapper.vm.$el.outerHTML
    expect($html).toMatchSnapshot()
  })

})
