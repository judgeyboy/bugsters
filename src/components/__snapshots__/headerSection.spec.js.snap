// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection Component should match snapshot 1`] = `"<div class=\\"header-section\\"><img src=\\"bugsters-logo.png\\" width=\\"150\\"> <h1>message.welcome</h1> <h2>Home of Professional Software Testers</h2> <a location=\\"about\\" class=\\"btn btn-primary\\">Read More</a></div>"`;
